src/ploomber/sources/notebooksource.py

```diff
@@ -606,6 +606,9 @@ def format(self, fmt):
         if extension_changed:
             path = self._path.with_suffix(ext_format)
             Path(self._path).unlink()
+            modified_entry = (Path(entry_point).read_text().replace(
```

Comment:
there's an edge case here. we also support other types of entry points, for example, it might be a path_to_module.path_to_function string so `read_text()` will fail in that case.

I suggest checking `Path(entry_point).is_file()`, if False, we should print a message with `click.secho` (in yellow font) saying that the entry point is not a file, and skip the extension replacement - the user will have to do it manually.

Reply:
Functions doesn't have format so it doesn't even gets to this code, it throws an exception
Out[1]: PythonCallableSource(function) (defined at: '/private/var/folders/qx/px2tf0rn5bq6c79yd3gk37fwzpvmb9/T/tmpgnur3y2z/content/my_tasks/raw/functions.py:4')
'PythonCallableSource' object has no attribute 'format' We can either implement it with the secho or print it as part of the except block

Reply:
I didn't mean a function which is a task in a pipeline. I meant an entry point. We support users defining a function that returns a DAG object. Example:

```
ploomber build --entry-point some_module.some_function
```

In such case, the code will break because entry_point isn't a file.

Example here: https://docs.ploomber.io/en/latest/user-guide/parametrized.html#Python-API-(factory-functions)